pennylane_lightning/core/src/utils/cuda_utils/DevicePool.hpp

```diff
@@ -28,7 +28,19 @@ namespace Pennylane::LightningGPU {
  */
 template <typename DeviceIndexType = int> class DevicePool {
   public:
-    DevicePool() {
+    DevicePool() { refresh(); }
+
+    DevicePool(DevicePool &&dp) { refresh(); }
```

Comment:
`[[maybe_unused]]` attribute tells the compiler that the `dp` might not be used and to suppress any warnings about unused argument at compilation.

```suggestion
    DevicePool([[maybe_unused]] DevicePool &&dp) { refresh(); }
```